app/Models/Interventions/Intervention.php

```
@@ -86,4 +89,30 @@ public function maintenanceContract(): BelongsTo
+
+    /**
+     * Renvoie le modèle de rapport applicable : celui lié à l'intervention
+     * (s'il est actif), sinon le plus récent modèle actif du même type.
+     */
+    public function applicableReportTemplate(): ?InterventionReportTemplate
+    {
+        if ($this->report_template_id) {
+            $linked = $this->reportTemplate()->where('is_active', true)->first();
+
+            if ($linked) {
+                return $linked;
```

PHPStan (2.2.7)

[error] Method App\Models\Interventions\Intervention::applicableReportTemplate() should return App\Models\Interventions\InterventionReportTemplate|null but returns Illuminate\Database\Eloquent\Model.


(return.type)